dttools/src/create_dir.h

```diff
@@ -10,20 +10,32 @@ See the file COPYING for details.
 
 /** @file create_dir.h Create a new directory recursively. */
 
+#include "mkdir_recursive.h"
+
+#include <sys/stat.h>
+
 /** Create a new directory recursively.
 @param path The full path of a directory.  It is not necessary for all components of the path to exist.
 @param mode The desired unix mode bits of the directory and parents.
 @return One on success, zero on failure.
 */
 
-int create_dir(const char *path, int mode);
+static inline int create_dir (const char *path, mode_t mode)
```

Comment:
Inlines and macros tend to cause confusion.

Unless there is some really good performance reason, let's just make these
ordinary functions that invoke the new functions.

(e.g. old binaries will still link correctly.)